### PR TITLE
[Windows] Fix FileManager.removeItem

### PR DESCRIPTION
**Pull request by gmittert**
- Get the fsr of the passed in path before using it
- Add the correct path to the directory stack

### CODE REVIEW

**Comment by gmittert** [comment]
@swift-ci please test

File: Foundation/FileManager+Win32.swift [hunk 1]
```diff
@@ -405,30 +405,37 @@ extension FileManager {
         guard alreadyConfirmed || shouldRemoveItemAtPath(path, isURL: isURL) else {
             return
         }
-        let faAttributes = try windowsFileAttributes(atPath: path)
+        let url = URL(fileURLWithPath: path)
+        var fsrBuf: UnsafeMutableBufferPointer<WCHAR> = UnsafeMutableBufferPointer.allocate(capacity: path.length * 2)
```
Comment (compnerd) [inline Foundation/FileManager+Win32.swift]: Let's use an array instead

Comment (gmittert) [reply Foundation/FileManager+Win32.swift]: done

**Comment by gmittert** [comment]
@swift-ci please test

**Comment by gmittert** [comment]
@swift-ci please test

File: Foundation/FileManager+Win32.swift [hunk 2]
```diff
@@ -405,30 +405,36 @@ extension FileManager {
         guard alreadyConfirmed || shouldRemoveItemAtPath(path, isURL: isURL) else {
             return
         }
-        let faAttributes = try windowsFileAttributes(atPath: path)
+        let url = URL(fileURLWithPath: path)
+        var fsrBuf: [WCHAR] = Array<WCHAR>(repeating: 0, count: Int(MAX_PATH))
+        _CFURLGetWideFileSystemRepresentation(url._cfObject, false, &fsrBuf, Int(MAX_PATH))
+        let length = fsrBuf.firstIndex(where: {$0 == 0}) ?? (path.length * 2)
```
Comment (compnerd) [inline Foundation/FileManager+Win32.swift]: This is probably better written as:

`let length = wcslen_s(&path, fsrBuf.count)`

**Comment by gmittert** [comment]
@swift-ci please test and merge

**Comment by gmittert** [comment]
@swift-ci please test and merge

**Comment by gmittert** [comment]
@swift-ci please test linux platform

**Comment by gmittert** [comment]
@swift-ci please test and merge

**Pull request merged**